DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json

```diff
@@ -0,0 +1,160 @@
+        {
+            "Title": "The index fields",
+            "Slug": "step-2",
+            "DescriptionLines": [
+                "* Define the index-fields.",
+                "",
+                "* i.e. _AttachmentNames_ will be configured to hold all attachments names (in step 4)."
+            ],
+            "Assets": [
+                {
+                    "Title": "Static Indexes Overview Demo",
+                    "Type" : "demo",
+                    "Url"  : "/static-indexes/static-indexes-overview"
+                },
+                {
+                    "Title": "Map Index Demo",
+                    "Type" : "demo",
+                    "Url"  : "/static-indexes/map-index"
+                }
+            ]
+        },
+        {
+            "Title": "Load details for indexing",
+            "Slug": "step-3",
+            "DescriptionLines": [
+                "* In the index Map function, call `AttachmentsFor()` to load the document's attachment details to be indexed.",
+                "",
+                "* The _attachments_ array will hold the following per attachment:",
+                "  * Name",
+                "  * Type",
+                "  * Hash",
+                "  * Size",
+                "",
+                "* To index an attachment content use: _LoadAttachment()_ or _LoadAttachments()_."
+            ],
+            "Assets": [
+                {
+                    "Title": "using AttachmentsFor",
+                    "Type" : "link",
+                    "Url"  : "https://ravendb.net/docs/article-page/latest/csharp/document-extensions/attachments/indexing#using-attachmentsfor"
+                },
+                {
+                    "Title": "using LoadAttachment",
+                    "Type" : "link",
+                    "Url"  : "https://ravendb.net/docs/article-page/latest/csharp/document-extensions/attachments/indexing#indexes-with--1"
+                },
+                {
+                    "Title": "using LoadAttachments",
+                    "Type" : "link",
+                    "Url"  : "https://ravendb.net/docs/article-page/latest/csharp/document-extensions/attachments/indexing#indexes-with--2"
+                }
+            ]
+        },
+        {
+            "Title": "Configure index fields",
+            "Slug" : "step-4",
+            "DescriptionLines": [
+                "* Configure the index-fields content.",
+                "",
+                "* The attachments names, types, hashes & sizes are now available and can be used to configure the index fields.<br>",
+                "  e.g., _AttachmentNames_ is an index field that holds all the attachments names of the document.",
+                "",
+                "* Later on, the 'Employees' collection can be queried using this index to get employees that have specific attachments names, types, etc.",
+                "",
+                "* Re-indexing will happen on any change to the document's attachments."
```

Comment:
So, indexing is smart enough to detect that only attachments are taken from the documents and changes to the document's JSON will not trigger reindexing for this index?

Reply:
yes - as far as I understand.
only changes made to the attachments will trigger indexing.
(that does include deleting a document that had attachments..)